### PR TITLE
Revert win2022 image back to latest and add msvc version mitigation

### PR DESCRIPTION
**Pull request by danieljurek**

### CODE REVIEW

File: eng/pipelines/templates/variables/image.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ variables:
   - name: LINUXNEXTVMIMAGE
     value: azsdk-pool-mms-ubuntu-2204-1espt
   - name: WINDOWSVMIMAGE
-    value: 1espt-win-2022-cpp-rollback
+    value: azsdk-pool-mms-win-2022-1espt
```
Comment (benbp) [inline eng/pipelines/templates/variables/image.yml:20]: After this merges can I remove this image?

File: eng/pipelines/templates/steps/setup-msvc.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+
+steps:
+  - pwsh: |
+      $tempFile = New-TemporaryFile;
+      systeminfo.exe /fo csv > $tempFile
+      $osName = (Import-Csv $tempFile).'OS Name'
+
+      Write-Host "OS Name: $osName"
+      if (!($osName -like '*2022*')) {
+        Write-Host "OS is not Windows Server 2022, skipping MSVC changes"
```
Comment (benbp) [inline eng/pipelines/templates/steps/setup-msvc.yml:10]: Do we not expect this problem to exist as well for future versions of windows server? I thought that it was a new issue introduced post-2019. If so I would think detecting whether there are multiple vs versions installed is a more robust method as opposed to just looking at OS versions.

File: eng/pipelines/templates/variables/image.yml (hunk 1 above)
Comment (danieljurek) [reply eng/pipelines/templates/variables/image.yml:20]: Yes. The image behind it was cleaned up by 1ES

File: eng/pipelines/templates/steps/setup-msvc.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+        exit 0
+      }
+
+      Remove-item 'C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Auxiliary\Build\Microsoft.VCToolsVersion.v143.default.*' -Force
+      Get-ChildItem 'C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Auxiliary\Build\' -Name | Write-Host
+    displayName: Setup MSVC
+    condition: eq(variables['Agent.OS'], 'Windows_NT')
```
Comment (weshaggard) [inline eng/pipelines/templates/steps/setup-msvc.yml:17]: should add in succeeded in the condition as well.

**Comment by benbp** [comment]
If this is a temp fix can we add some TODO comments and a follow-up issue so this doesn't stick around in source?

**Review by weshaggard — APPROVED** [review]
I'm good with this for a temporary fix.

**Review by mikeharder — APPROVED** [review]

File: eng/pipelines/templates/steps/setup-msvc.yml (hunk 2 above)
Comment (danieljurek) [reply eng/pipelines/templates/steps/setup-msvc.yml:10]: We expect this problem to be gone circa mid-May: https://github.com/actions/runner-images/issues/9701

**Review by antkmsft — APPROVED** [review]

**Review by RickWinter — APPROVED** [review]

**Pull request merged**